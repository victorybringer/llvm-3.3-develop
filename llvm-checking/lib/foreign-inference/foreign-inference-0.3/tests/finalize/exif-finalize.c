#include <stdlib.h>
struct Mem {
  void* (*alloc_func)(size_t);
  void (*free_func)(void*);
};

struct Mem* mem_new(void* (*alloc)(size_t), void (*dealloc)(void*)) {
  struct Mem* m = malloc(sizeof(struct Mem));
  m->alloc_func = alloc;
  m->free_func = dealloc;

  return m;
}

struct Mem* mem_new_default() {
  return mem_new(malloc, free);
}

void exif_mem_free(struct Mem *mem, void *d) {
  mem->free_func(d);
}
