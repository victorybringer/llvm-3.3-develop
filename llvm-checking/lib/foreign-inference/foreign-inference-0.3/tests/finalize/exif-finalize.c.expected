fromList [ ("@exif_mem_free", fromList ["%d"]) ]